src/Entity/ApiDoc.php

```diff
@@ -332,6 +333,23 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       ->setLabel(t('Spec fetched from URL timestamp'))
       ->setDescription(t('When the OpenAPI spec file was last fetched from URL as a Unix timestamp.'));
 
+    // Gets whether the access control should be configurable per entity.
+    $product_access_is_configurable = (\Drupal::config(ApiDocSettingsForm::CONFIG_NAME)->get('enable_product_access_control') === 'configurable');
+    // Store weather product access should be checked per entity.
+    $fields['product_access_control'] = BaseFieldDefinition::create('boolean')
+      ->setLabel(t('Restrict access based on API product'))
+      ->setDescription(t('If checked, access to view this API doc will restricted to users who have access to the associated API product.'))
```

Comment:
Should add a link here also to the API product access settings so that user understands where the API product access control settings are:
```suggestion
      ->setDescription(t('If checked, access to view this API doc will restricted using the API product access control settings.'))
```